File: android-app/src/main/kotlin/com/jworks/kanjiquest/android/ui/navigation/NavRoutes.kt
package com.jworks.kanjiquest.android.ui.navigation

sealed class NavRoute(val route: String) {
    data object Splash : NavRoute("splash")
    data object Login : NavRoute("login")
    data object Home : NavRoute("home")
    data object KanjiDetail : NavRoute("kanji/{kanjiId}") {
        fun createRoute(kanjiId: Int) = "kanji/$kanjiId"
    }
    data object Recognition : NavRoute("game/recognition")
    data object Writing : NavRoute("game/writing")
    data object Vocabulary : NavRoute("game/vocabulary")
    data object Camera : NavRoute("game/camera")
    data object Shop : NavRoute("shop")
    data object Progress : NavRoute("progress")
    data object Settings : NavRoute("settings")
    data object Achievements : NavRoute("achievements")
    data object Subscription : NavRoute("subscription")
    data object WritingTargeted : NavRoute("game/writing/{kanjiId}") {
        fun createRoute(kanjiId: Int) = "game/writing/$kanjiId"
    }
    data object RecognitionTargeted : NavRoute("game/recognition/{kanjiId}") {
        fun createRoute(kanjiId: Int) = "game/recognition/$kanjiId"
    }
    data object VocabularyTargeted : NavRoute("game/vocabulary/{kanjiId}") {
        fun createRoute(kanjiId: Int) = "game/vocabulary/$kanjiId"
    }
    data object CameraTargeted : NavRoute("game/camera/{kanjiId}") {
        fun createRoute(kanjiId: Int) = "game/camera/$kanjiId"
    }
    data object Flashcards : NavRoute("flashcards")
    data object FlashcardStudy : NavRoute("flashcards/study/{deckId}") {
        fun createRoute(deckId: Long) = "flashcards/study/$deckId"
    }
    data object DevChat : NavRoute("devchat")
    data object PlacementTest : NavRoute("placement_test")
    data object WordDetail : NavRoute("word/{wordId}") {
        fun createRoute(wordId: Long) = "word/$wordId"
    }
    data object KanaRecognition : NavRoute("game/kana_recognition/{kanaType}") {
        fun createRoute(kanaType: String) = "game/kana_recognition/$kanaType"
    }
    data object KanaWriting : NavRoute("game/kana_writing/{kanaType}") {
        fun createRoute(kanaType: String) = "game/kana_writing/$kanaType"
    }
    data object RadicalDetail : NavRoute("radical/{radicalId}") {
        fun createRoute(radicalId: Int) = "radical/$radicalId"
    }
    data object RadicalRecognition : NavRoute("game/radical_recognition")
    data object RadicalBuilder : NavRoute("game/radical_builder")
    data object FieldJournal : NavRoute("field_journal")
    data object Collection : NavRoute("collection")
    data object Study : NavRoute("study")
    data object Games : NavRoute("games")
    data object CollectionHub : NavRoute("collection_hub")
    data object TestMode : NavRoute("game/test_mode")
}

File: ios-app/KanjiQuest/Navigation/NavRoutes.swift
import Foundation

/// All navigation routes matching the Android NavRoutes sealed class.
/// Uses associated values for parameterized routes.
enum NavRoute: Hashable {
    // Core
    case splash
    case login
    case home
    case placementTest

    // Kanji detail
    case kanjiDetail(kanjiId: Int32)

    // Game modes
    case recognition
    case recognitionTargeted(kanjiId: Int32)
    case writing
    case writingTargeted(kanjiId: Int32)
    case vocabulary
    case vocabularyTargeted(kanjiId: Int32)
    case camera
    case cameraTargeted(kanjiId: Int32)

    // Kana modes
    case kanaRecognition(kanaType: String)
    case kanaWriting(kanaType: String)

    // Radical modes
    case radicalRecognition
    case radicalBuilder
    case radicalDetail(radicalId: Int32)

    // Collection & Flashcards
    case collection
    case flashcards
    case flashcardStudy(deckId: Int64)

    // Progress & Achievements
    case progress
    case achievements

    // Shop & Subscription
    case shop
    case subscription

    // Settings
    case settings

    // Word detail
    case wordDetail(wordId: Int64)

    // Test Mode (Games hub)
    case testMode

    // Dev & Feedback
    case devChat
    case fieldJournal

    // iPad-only calligraphy
    #if IPAD_TARGET
    case calligraphySession(kanjiLiteral: String, strokePaths: [String])
    #endif
}
